backend/src/api/solutions/solutions.service.ts

```diff
@@ -64,9 +95,44 @@ export class SolutionsService {
       task: { connect: { id: taskId } },
       sessionTask: { connect: { sessionId_taskId: { sessionId, taskId } } },
     };
-    return this.prisma.solution.create({
+
+    const solution = await this.prisma.solution.create({
       data: checkedSolution,
-      omit: omitData,
     });
+
+    // perform the analysis but do *not* wait for the promise to resolve
+    // this will happen in the background
+    this.analysisService.performAnalysis(solution);
+
+    return solution;
+  }
+
+  // check every minute (with seconds = 0) whether there are analyses that were not performed
+  @Cron("0 * * * * *", { name: "runUnperformedAnalyes" })
```

Comment:
any reason to do it on a minutely-basis and not at higher frequency ?

Reply:
Nope, what did you have in mind?